packages/design-system/src/components/Button/Button.tsx

```diff
@@ -147,6 +145,12 @@ export const Button = ({
     // and remove <button> specific attributes
     delete attrs.disabled;
     delete attrs.type;
+
+    if (disabled) {
+      attrs.role = 'link';
+      attrs['aria-disabled'] = true;
+      delete attrs.href;
+    }
```

Comment:
[Discussed with @pwolfert and @SparrowSpit in Slack.](https://adhoc.slack.com/archives/C645GLLNN/p1659032212248209)

To summarize, we're adding better support for disabled anchor buttons. 

The way this works is users still pass an `href` prop to the `<Button>` to indicate if they want a `<button>` or an `<a>` element. However, if `disabled` is also passed (`<Button href="#!" disabled />`), it will strip the `href` attr from the anchor element and pass additional a11y-friendly attrs.

The disabled class logic was also removed as we're relying on `aria-disabled` to properly style our disabled link buttons.